lib/rpm_package.py

```diff
@@ -15,16 +15,81 @@
 
 import os
 import logging
+import re
 import yaml
 
 from lib import config
 from lib import exception
+from lib import utils
 from lib.package import Package
 
 CONF = config.get_config().CONF
 LOG = logging.getLogger(__name__)
 
 
+def compare_versions(v1, v2):
+    try:
+        utils.run_command("rpmdev-vercmp %s %s" % (v1, v2))
+        rc = 0
+    except exception.SubprocessError as exc:
+        if exc.returncode == 11:
+            rc = 1
+        elif exc.returncode == 12:
+            rc = -1
+        else:
+            raise
+    return rc
+
+
+class SpecFile(object):
+
+    def __init__(self, path):
+        self.path = path
+
+    def query_tag(self, tag):
+        return utils.run_command(
+            "rpmspec --srpm -q --qf '%%{%s}' %s 2>/dev/null" % (
+                tag.upper(), self.path)).strip()
+
+    def update_version(self, new_version):
+        LOG.info("Updating '%s' version to: %s" % (self.path, new_version))
+
+        with open(self.path, 'r+') as f:
+            content = f.read()
```

Comment:
Opening the file here is not optimal. I'm adding code to update the 'prerelease' tag and this will be duplicated. Perhaps we should have `read_spec` and `write_spec` methods and keep the content of the file in an attribute so that we read it and write it only once. We could also delay the reading until a functions needs it. What do you think?